src/models/state.js

```diff
@@ -426,6 +426,22 @@ class State extends new Record(DEFAULTS) {
   }
 
   /**
+   * Check whether the selection is empty.
+   *
+   * @return {Boolean}
+   */
+
+  get isEmpty() {
+    const { startKey, endKey, startOffset, endOffset } = this
+
+    if (startKey == endKey) {
+      return (startOffset == endOffset)
+    }
+
+    return this.fragment.text.length == 0
```

Comment:
And continuing on this idea, you can afterward check if the other offset is at the end of its node.